.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,154 @@
+      id: cache
+
+
+    - name: Create conda environment from environment.yml
+      # Update the tudat conda environment created using the environment.yml file if the cache is not restored
+      run: mamba env update -n tudat -f environment.yml
+      if: steps.cache.outputs.cache-hit != 'true'
+
+
+    - name: Download tudat resources data files for Linux and MacOS
+      # The data files in tudat-resources are downloaded upon the installation of
+      # the tudat-resources conda package in the previous step, but the data files are
+      # not cached by the cache action. When the cache is hit, the update environment
+      # step is skipped, skipping the installation of the tudat-resources package,
+      # thereby skipping the download of the data files. Therefore, the data files must
+      # be downloaded manually when the cache is hit.
+      # This step downloads the data files in the tudat-resources package by running
+      # the post-link scripts in https://github.com/tudat-team/tudat-resources-feedstock/blob/master/recipe/
+      if: (runner.os == 'Linux' || runner.os == 'MacOS') && steps.cache.outputs.cache-hit == 'true'
+      run: bash -c "$(curl -fsSL https://raw.githubusercontent.com/tudat-team/tudat-resources-feedstock/master/recipe/post-link.sh)"
+
+
+    - name: Download tudat-resources data files for Windows
+      if: runner.os == 'Windows' && steps.cache.outputs.cache-hit == 'true'
+      run: cmd /c "curl -fsSL https://raw.githubusercontent.com/tudat-team/tudat-resources-feedstock/master/recipe/post-link.bat | cmd"
+
+
+    - name: Restore cached compilation results
+      # This third party action will setup ccache on the runner and restore any previously saved cache. Caches are saved automatically by the action after a build.
+      uses: hendrikmuhs/ccache-action@v1.2.12
+      with:
+        key: ${{ matrix.os }}-${{ matrix.build_type }}
+
+
+    - name: Configure CMake
+      # Configure CMake in a 'build' subdirectory. `CMAKE_BUILD_TYPE` is only required if you are using a single-configuration generator such as make.
+      # See https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html?highlight=cmake_build_type
+      shell: bash -l {0}
+      run: >
+        cmake -B "${{ github.workspace }}/build"
+        -S "${{ github.workspace }}"
+        -DCMAKE_BUILD_TYPE=${{ matrix.build_type }}
+        -DCMAKE_CXX_COMPILER=${{ matrix.cpp_compiler }}
+        -DCMAKE_CXX_COMPILER_LAUNCHER=ccache
+        -DCMAKE_C_COMPILER=${{ matrix.c_compiler }}
+        -DCMAKE_C_COMPILER_LAUNCHER=ccache
+        -DTUDAT_BUILD_GITHUB_ACTIONS=ON
+
+
+    - name: Build
+      # Build your program with the given configuration. Note that --config is needed because the default Windows generator is a multi-config generator (Visual Studio generator). Use multiple cpu cores available in the runner.
+      shell: bash -l {0}
+      run: cmake --build "${{ github.workspace }}/build" --config "${{ matrix.build_type }}" -j4
```

Comment:
I am not sure if this would cause any problems but according to this: https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories `macos-latest` only has 3 cores available if I understand correctly.

Reply:
Thanks for the catch @LeonardoPohl :) !  We had a look and it appears the [Apple M1s have 8 cores each](https://en.wikipedia.org/wiki/Apple_M1), giving us a total of 24 cores on the runner.